bson/src/main/scala/types.scala

```diff
@@ -379,63 +379,13 @@ class BSONObjectID private (private val raw: Array[Byte])
 
 object BSONObjectID {
   private val maxCounterValue = 16777216
-  private val increment = new java.util.concurrent.atomic.AtomicInteger(scala.util.Random.nextInt(maxCounterValue))
+  private val secureRandom = new java.security.SecureRandom()
+  private val increment = new java.util.concurrent.atomic.AtomicInteger(secureRandom.nextInt(maxCounterValue))
+  private val randomVal1 = secureRandom.nextInt(0x01000000)
```

Comment:
There's no need to put these here, they could be limited to the scope where they are used.